.github/workflows/post-merge-tests-and-tag.yml

```diff
@@ -0,0 +1,33 @@
+name: Run Tests after Pull Request is merged and Tag
+
+on:
+  push:
+    branches:
+      - main
+
+permissions:
+  contents: write
+
+jobs:
+  run-tests-and-tag:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683
+        with:
+          fetch-depth: 0
+
+      - name: Run tests
+        uses: ./.github/workflows/run-tests.yaml
```

Comment:
I've just noticed that you're including `run-tests.yaml` as a `step`, but the file you're including defines jobs. I'm not sure what GHA will do with this, but it may just error. If you look at `run-tests-and-push-to-ghcr-on-main.yml` you'll see that the file is included as a separate job, rather than a step.

I think we need to restructure this slightly so that we have separate jobs for `Run tests` and `Tag release`. This will also allow setting the write permissions in the `Tag release` job only, so that the tests only run with read perms.

There's another example of how to structure this [here](https://github.com/alphagov/pay-connector/blob/master/.github/workflows/post-merge.yml).

Reply:
I've separated this into two jobs and fixed the permissions in https://github.com/alphagov/re-request-an-aws-account/pull/388/commits/c84c2f56380a9e33eca92191d64c12dc3281540e